### PR TITLE
feature/ICS-134 persist rejection reason

### PR DESCRIPTION
**Pull request by JGiter**
https://energyweb.atlassian.net/browse/ICS-134

### CODE REVIEW

File: src/modules/claim/claim.types.ts [hunk 1]
```diff
@@ -41,6 +41,13 @@ export enum RegistrationTypes {
   OnChain = 'RegistrationTypes::OnChain',
 }
 
+export enum RejectionReason {
+  ALREADY_ISSUED = 'Email already associated with DID',
+  MAIL_IS_BLACKLISTED = 'Email belongs to blacklisted domain',
+  CONFIRMATION_LINK_EXPIRED = 'The email confirmation link has expired',
+  VERIFICATION_SEND_ERROR = 'Unable to send verification email to provided email address',
+}
+
```
Comment (jrhender) [inline src/modules/claim/claim.types.ts:50]: I don't think this is necessary. The issuer/rejector should be able to put whatever they want
```suggestion
```

File: src/modules/claim/claim.dto.ts [hunk 2]
```diff
@@ -157,6 +158,10 @@ export class ClaimRejectionDTO implements IClaimRejection {
   @IsBoolean()
   @ApiProperty()
   isRejected: boolean;
+
+  @IsEnum(RejectionReason)
```
Comment (jrhender) [inline src/modules/claim/claim.dto.ts:162]: Let's keep it simple. Maybe a max-length though would be good though. Maybe 200 char or something?
```suggestion
  @IsString()
```

File: src/modules/claim/claim.types.ts [hunk 3]
```diff
@@ -54,6 +61,7 @@ export interface IRoleClaim {
   isAccepted: boolean;
   acceptedBy?: string;
   isRejected?: boolean;
+  rejectionReason?: RejectionReason;
```
Comment (jrhender) [inline src/modules/claim/claim.types.ts:64]:
```suggestion
  rejectionReason?: string;
```

File: src/modules/claim/entities/roleClaim.entity.ts [hunk 4]
```diff
@@ -74,6 +74,10 @@ export class RoleClaim implements IRoleClaim {
   @Column({ nullable: true, type: 'bool', default: false })
   isRejected?: boolean | null;
 
+  @Field({ nullable: true })
+  @Column({ nullable: true, type: 'enum', enum: RejectionReason })
+  rejectionReason?: RejectionReason;
```
Comment (jrhender) [inline src/modules/claim/entities/roleClaim.entity.ts:79]: change to string

**Review by jrhender — CHANGES_REQUESTED** [review]
I think we should keep it simple and let the requestor/issuer define the rejection reason. Not only does it simplify the code, it isn't realistic to have to add values to the `enum` for every possible reason for every possible use case

**Comment by jrhender** [comment]
@JGiter why are there 295 files changed? Is this expected?

**Comment by JGiter** [comment]
> @JGiter why are there 295 files changed? Is this expected?

Probably these .md left from previous build. Cleared them

**Review by jrhender — APPROVED** [review]
Looks good to me, thanks @JGiter 😄

**Pull request merged**